src/referencedata-facility/facility.service.js

```diff
@@ -307,7 +307,16 @@
              * @return {Promise} Array of facilities with minimal representation
              */
             function getAllMinimal() {
-                return resource.getAllMinimal().$promise;
+                var deferred = $q.defer();
+
+                resource.getAllMinimal(function(facilities) {
+                    var sortedFacilities = _.sortBy(facilities, 'name');
```

Comment:
I don't think this is the responsibility of the facilityService. It probably should be done on the server side.